ignite/pkg/cosmosgen/generate_typescript.go

```diff
@@ -121,90 +132,66 @@ func (g *tsGenerator) generateModuleTemplates() error {
 		}
 	}
 
-	add(g.g.appPath, g.g.appModules)
+	add(g.g.appPath, g.g.appModules, g.g.appIncludes)
 
 	// Always generate third party modules; This is required because not generating them might
 	// lead to issues with the module registration in the root template. The root template must
 	// always be generated with 3rd party modules which means that if a new 3rd party module
 	// is available and not generated it would lead to the registration of a new not generated
 	// 3rd party module.
 	for sourcePath, modules := range g.g.thirdModules {
-		add(sourcePath, modules)
+		includes := g.g.thirdModuleIncludes[sourcePath]
+		add(sourcePath, modules, append(g.g.appIncludes, includes...))
 	}
 
 	return gg.Wait()
 }
 
 func (g *tsGenerator) generateModuleTemplate(
 	ctx context.Context,
+	protocCmd protoc.Cmd,
 	staCmd sta.Cmd,
-	appPath string,
+	tsprotoPluginPath, appPath string,
 	m module.Module,
+	includePaths []string,
 ) error {
 	var (
 		out      = g.g.o.jsOut(m)
 		typesOut = filepath.Join(out, "types")
-		conf     = swaggercombine.Config{
-			Swagger: "2.0",
-			Info: swaggercombine.Info{
-				Title: "HTTP API Console",
-			},
-		}
 	)
-
 	if err := os.MkdirAll(typesOut, 0o766); err != nil {
 		return err
 	}
 
 	// generate ts-proto types
-	if err := g.g.buf.Generate(
+	err := protoc.Generate(
 		ctx,
-		m.Pkg.Path,
 		typesOut,
-		g.tsTemplate(),
-		"module.proto",
-	); err != nil {
-		return err
-	}
-
-	// generate OpenAPI spec
-	tmp, err := os.MkdirTemp("", "gen-js-openapi-module-spec")
+		m.Pkg.Path,
+		includePaths,
+		tsOut,
+		protoc.Plugin(tsprotoPluginPath, "--ts_proto_opt=snakeToCamel=true", "--ts_proto_opt=esModuleInterop=true"),
```

Comment:
can we use the `experimental_allow_proto3_optional` flag with the `protoc`?

https://github.com/ignite/cli/issues/3698

Reply:
we can test it...not sure how the ts proto plugin behaves